src/main/scala/com/microsoft/hyperspace/actions/Action.scala

```diff
@@ -33,6 +33,7 @@ import com.microsoft.hyperspace.telemetry.{AppInfo, HyperspaceEvent, HyperspaceE
  */
 trait Action extends HyperspaceEventLogging with Logging with ActiveSparkSession {
   protected val baseId: Int = logManager.getLatestId().getOrElse(-1)
+  protected def endId: Int = baseId + 2
```

Comment:
Can you mark this as `final`?

Reply:
Can you also update `end()` to use this method for the `newId`?

Reply:
I introduced this because of #272 since `id` is not available in `getLogEntry` when IndexLogEntry creation.
However, just for this PR, it looks unnecessary. I removed this.